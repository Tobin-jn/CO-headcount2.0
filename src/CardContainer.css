.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  background-color: #6f9d33;
}