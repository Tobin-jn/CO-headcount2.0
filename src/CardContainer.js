import React from 'react';
import Card from './Card';
import './CardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = (props) => {

  const cards = props.data.map(district => {
    return <Card 
      district={district} 
      key={Math.random()} 
      compareDistrictData={props.compareDistrictData} />;
  });
  
  return (
    <div className='card-container'>
      {cards}
    </div>
  );
};

CardContainer.propTypes = {
  data: PropTypes.array,
  compareDistrictData: PropTypes.func
};

export default CardContainer;