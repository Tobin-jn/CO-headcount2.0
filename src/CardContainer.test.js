import React from 'react'
import { shallow, mount } from 'enzyme'
import CardContainer from './CardContainer'

describe('CardContainer', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const mockStats = {stats: {location: [{year: 2007}]}}
    const wrapper = shallow(<CardContainer stats={mockStats} />)
    expect(wrapper).toMatchSnapshot()
  })
})