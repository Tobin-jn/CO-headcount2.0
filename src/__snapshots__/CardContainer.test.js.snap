// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card-container"
>
  <Card
    compareDistrictData={[Function]}
    district={
      Object {
        "location": "denver",
        "stats": Array [
          Object {
            "2001": 2,
          },
        ],
      }
    }
    key="0.19449928746916045"
  />
  <Card
    compareDistrictData={[Function]}
    district={
      Object {
        "location": "boulder",
        "stats": Array [
          Object {
            "2002": 1,
          },
        ],
      }
    }
    key="0.2592622780689873"
  />
</div>
`;
